src/main/scala/akka/stream/contrib/Sample.scala

```diff
@@ -0,0 +1,64 @@
+package akka.stream.contrib
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+
+object Sample
+{
+  /**
+    *
+    * returns every nth elements
+    *
+    * @param nth
+    * @tparam T
+    * @return
+    */
+  def apply[T](nth: Int): Sample[T] = Sample[T](() => nth)
+
+  // def random[T](maxStep: Int = 1000, random: Random = Random): Sample[T] = Sample[T](() => random.nextInt(maxStep))
```

Comment:
Alternative is to take a function `() => Int`, since that does not tie the random sampling to a particular interface. Then the default can point to a cached function using a random generator (possibly `ThreadLocalRandom`, although these are very low quality random numbers).